src/Microsoft.VisualStudio.Services.Agent/SecretMask.cs

```diff
@@ -0,0 +1,82 @@
+using System;
+using System.Collections.Generic;
+using System.Text.RegularExpressions;
+
+namespace Microsoft.VisualStudio.Services.Agent
+{
+    public interface ISecretMask
+    {
+        /// <summary>
+        /// Returns one item (start, length) for each match found in the input string.
+        /// </summary>
+        /// <param name="input"></param>
+        /// <returns></returns>
+        IEnumerable<Tuple<int, int>> GetPositions(string input);
```

Comment:
Can we make a class instead of tuple? I now remember reading secret masker code several months back and stumbling over readability of tuple.

I see there's already a ReplacementPosition class down below as well. It looks like that could be used instead. Also consider shortening it's name to ReplacePosition or SecretPosition or MaskPosition or something.